blueocean-dashboard/src/main/js/components/karaoke/services/pagers/PipelinePager.js

```diff
@@ -114,8 +114,19 @@ export class PipelinePager {
                         // Actually we should only come here on a not running job
                         logger.debug('Actually we should only come here on a not running job');
                         const convertedNodes = convertJenkinsNodeGraph(logData.data.model, result.isFinished, 0);
-                        const lastNode = logData.data.model.filter(node => node.id === convertedNodes[convertedNodes.length - 1].id)[0];
+                        let lastConvertedNode = convertedNodes[convertedNodes.length - 1];
 
+                        //select the last child if the node has any children
+                        if (lastConvertedNode.children.length) {
+                            lastConvertedNode = lastConvertedNode.children[lastConvertedNode.children.length - 1];
+
+                            //if the node has siblings, select the last sibling
+                            while (lastConvertedNode.nextSibling) {
```

Comment:
Api returns a cyclic graph so probably it shouldn’t result in to infinite loop. Should it put any check to prevent it, just in case?